src/otx/engine/engine.py

```diff
@@ -269,12 +273,17 @@ def train(
             logging.warning(msg)
             self.model.label_info = self.datamodule.label_info
 
-        if resume:
-            fit_kwargs["ckpt_path"] = self.checkpoint
-        elif self.checkpoint is not None:
-            loaded_checkpoint = torch.load(self.checkpoint)
-            # loaded checkpoint have keys (OTX1.5): model, config, labels, input_size, VERSION
-            self.model.load_state_dict(loaded_checkpoint)
+        if resume and checkpoint:
+            # NOTE: If both `resume` and `checkpoint` are provided,
+            # load the entire model state from the checkpoint using the pl.Trainer's API.
+            fit_kwargs["ckpt_path"] = checkpoint
+        elif not resume and checkpoint:
+            # NOTE: If `resume` is not enabled but `checkpoint` is provided,
+            # load the model state from the checkpoint incrementally.
+            # This means only the model weights are loaded. If there is a mismatch in label_info,
+            # perform incremental weight loading for the model's classification layer.
+            ckpt = torch.load(checkpoint)
+            self.model.load_state_dict_incrementally(ckpt)
```

Comment:
I think we need to guide user about this logic. Can we add warning or documentation for this?

Reply:
I'll revisit the documentation in the future.